### DISCUSSION TITLE
How to add `contentOnly` editing support to a custom block

### DISCUSSION THREAD

**Discussion by ndiego** [WordPress/developer-blog-content#300] [Topic Ideas]
A lot of work is being done on `contentOnly` editing. While this functionality has been around for a while, upcoming enhancements will make this editing mode commonplace, especially among new users of WordPress. Here are a few related issues/PRs:

- https://github.com/WordPress/gutenberg/issues/60021#issuecomment-2325407240
- https://github.com/WordPress/gutenberg/pull/64923

While work is being done to ensure Core blocks work as expect within this editing mode, block developers will need to make modifications to their custom blocks if they want to support this mode. 

This article will detail what `contentOnly` editing is, how to apply it to blocks, and how to modify custom blocks to support `contentOnly` editing. 

cc @getdave @noisysocks in case you guys have any thoughts on this article topic.

**Comment by fabiankaegy** [comment]
I would be great to have a stable official way of doing this. 

Today we can do it by adding `__experimentalRole: content` to an attribute but it isn't documented and has the `__experimental` prefix...

**Reply by ndiego** [reply]
Yeah, I know 😬. If `contentOnly` becomes ubiquitous as intended by the link issues/PRs, we need to stabilize that property. There is also a fair amount of customization you need to do in a custom plugin to "gracefully" support `contentOnly`.

**Reply by ndiego** [reply]
I explored this in my Icon Block. It hasn't been released in the public version of the plugin yet, but it took some work: https://github.com/ndiego/icon-block/commit/cacd65bd18dfcd3d7676d3f8d865c06b660b5202

**Reply by fabiankaegy** [reply]
Yeah sorry this was not to imply that you don't know it :) 

This will also be even more interesting when we actually find a way to enable custom block bindings for pattern overrides :) (https://github.com/WordPress/gutenberg/issues/64870) I know it is somewhat different but also kind of the same  😂

**Reply by ndiego** [reply]
> Yeah sorry this was not to imply that you don't know it :)

No worries, I didn't take it that way! I created this discussion after seeing all the `contentOnly` updates and knowing that the experience in custom blocks was not great and would require additional guidance.

**Reply by noisysocks** [reply]
Yes, we should stabilise and document this.

**Comment by bph** [comment]
This topic as approved in 9/5 Editorial Group meeting. Please continue discussion on the issue
- #300 303

**Comment by ndiego** [comment]
Post has been published: https://developer.wordpress.org/news/2024/11/05/how-to-add-content-only-editing-support-to-a-block/
